Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -248,15 +251,24 @@ open class ChatChannelListVC: _ViewController,
         self.controller.synchronize()
         reloadChannels()
     }
-
+    
     /// Updates the list view with the most updated channels.
     public func reloadChannels() {
+        if isApplyingChangesets {
+            DispatchQueue.main.async {
+                self.reloadChannels()
+            }
+            return
+        }
+        
+        isApplyingChangesets = true
         let previousChannels = channels
         let newChannels = Array(controller.channels)
         let stagedChangeset = StagedChangeset(source: previousChannels, target: newChannels)
-        collectionView.reload(using: stagedChangeset, reconfigure: { _ in false }) { [weak self] newChannels in
+        collectionView.reload(using: stagedChangeset, reconfigure: { _ in true }) { [weak self] newChannels in
             self?.channels = newChannels
         }
+        isApplyingChangesets = false
```

Comment:
Shouldn't this be set inside the reload completion block? 🤔

Reply:
I also think that

Reply:
This is actually not a completion but it is the `setData` closure which is called before `performBatchUpdates` (too early).

```swift
 func reload<C>(
        using stagedChangeset: StagedChangeset<C>,
        reconfigure: ((IndexPath) -> Bool)? = nil,
        interrupt: ((Changeset<C>) -> Bool)? = nil,
        setData: (C) -> Void
    ) {
```

where `setData` is called after each changeset

```swift
for changeset in stagedChangeset {
            if let interrupt = interrupt, interrupt(changeset), let data = stagedChangeset.last?.data {
                setData(data)
                return reloadData()
            }
            
            performBatchUpdates({
                setData(changeset.data)
                
                if !changeset.sectionDeleted.isEmpty {
                    deleteSections(IndexSet(changeset.sectionDeleted))
                }
                // later we call reconfigureItems which can trigger re-entrancy
```